packages/medusa-js/src/request.ts

```diff
@@ -103,9 +110,16 @@ class Client {
     }
 
     // TODO: if route is an authenticated route, add api key
-    if (path.startsWith("/admin")) {
+    if (
+      path.startsWith("/admin") &&
+      !(
+        method === "POST" &&
+        typeof unAuthenticatedAdminEndpoints[path] !== "undefined"
+      )
+    ) {
```

Comment:
nit: Think we could increase readability on this one. Would it make sense to extract the entire condition to a function and include allowed methods to endpoints in above mapping (`unAuthenticatedAdminEndpoints`)? Let me know, what you think.

Would also allow us to easily include authenticated `/store/customers` endpoints when we've nailed cookie-based auth.